Detectors/ITSMFT/ITS/tracking/src/TrackerTraits.cxx

```diff
@@ -821,8 +821,14 @@ bool TrackerTraits::fitTrack(TrackITSExt& track, int start, int end, int step, f
       return false;
     }
 
-    if (!propInstance->propagateToX(track, trackingHit.xTrackingFrame, getBz(), o2::base::PropagatorImpl<float>::MAX_SIN_PHI, o2::base::PropagatorImpl<float>::MAX_STEP, mCorrType)) {
-      return false;
+    if (useBxByBz) {
+      if (!propInstance->PropagateToXBxByBz(track, trackingHit.xTrackingFrame, o2::base::PropagatorImpl<float>::MAX_SIN_PHI, o2::base::PropagatorImpl<float>::MAX_STEP, mCorrType)) {
+        return false;
+      };
+    } else {
+      if (!propInstance->propagateToX(track, trackingHit.xTrackingFrame, getBz(), o2::base::PropagatorImpl<float>::MAX_SIN_PHI, o2::base::PropagatorImpl<float>::MAX_STEP, mCorrType)) {
+        return false;
+      };
```

Comment:
Actually, there is this method which takes the switch:
https://github.com/AliceO2Group/AliceO2/blob/82f5a7fefde6c60de00fc95e3c370d96d95698e0/Detectors/Base/include/DetectorsBase/Propagator.h#L92

Reply:
thanks